packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -439,14 +456,16 @@ export default function () {
           }
 
           // add __esModule declaration if this file has any exports
-          if (hasExports && !strict) {
+          if (hasExports) {
             let buildTemplate = buildExportsModuleDeclaration;
             if (this.opts.loose) buildTemplate = buildLooseExportsModuleDeclaration;
 
-            const declar = buildTemplate();
+            const declar = buildTemplate(t.arrayExpression(Object.keys(exportNames).map(function (name) {
```

Comment:
I'd recommend

```
buildTemplate({
  NAMED_EXPORTS: t.valueToNode(Object.keys(exportNames)),
});
```

to use a name rather than `$0` and `t.valueToNode` to make the array into AST nodes.